dotcom-rendering/src/components/ListenToArticle.importable.tsx

```diff
@@ -0,0 +1,49 @@
+import { log } from '@guardian/libs';
+import { useEffect, useState } from 'react';
+import { getListenToArticleClient } from '../lib/bridgetApi';
+import { useIsBridgetCompatible } from '../lib/useIsBridgetCompatible';
+import { ListenToArticleButton } from './ListenToArticleButton';
+
+type Props = {
+	articleId: string;
+};
+export const ListenToArticle = ({ articleId }: Props) => {
+	const [showButton, setShowButton] = useState<boolean>(false);
+
+	const isBridgetCompatible = useIsBridgetCompatible('8.5.1');
+
+	useEffect(() => {
+		if (isBridgetCompatible) {
+			void getListenToArticleClient()
+				.isAvailable(articleId)
+				.then(() => setShowButton(false));
+			void getListenToArticleClient()
+				.isPlaying(articleId)
+				.then(() => setShowButton(false));
```

Comment:
I may be missing something, but I can't see anywhere in this component that `setShowButton` is set to `true`, which would mean it never renders? As mentioned above, I think `Promise.all`[^1] may be helpful here, as it allows you to make two asynchronous requests to the native layer, wait for both to complete, and then decide whether to show the button:

```suggestion
			Promise.all([
				getListenToArticleClient().isAvailable(articleId),
				getListenToArticleClient().isPlaying(articleId),
			]).then(([isAvailable, isPlaying]) =>
				setShowButton(isAvailable && !isPlaying),
			);
```

[^1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all